libvips/resample/templates.h

```diff
@@ -473,10 +476,41 @@ vips_reduce_make_mask(T *c, VipsKernel kernel, const int n_points,
 	}
 }
 
+/* Machinery to pick an intermediate for type T, prevents an overflow in
+ * reduce_sum(). Defaults to a 32-bit integral type.
+ */
+template <typename T,
+	bool IsIntegral = std::is_integral<T>::value,
+	size_t Size = sizeof(T)>
+struct intermediate {
+	typedef int32_t type;
+};
+
+/* 32-bit integral types needs a 64-bits intermediate.
+ */
+template <typename T>
+struct intermediate<T, true, 4> {
+	typedef int64_t type;
+};
+
+/* 32-bit floating-point types needs a 64-bits intermediate.
+ */
+template <typename T>
+struct intermediate<T, false, 4> {
+	typedef double type;
+};
+
+/* 64-bit floating-point types needs a 128-bits intermediate.
+ */
+template <typename T>
+struct intermediate<T, false, 8> {
```

Comment:
If `long double` is unavailable should we use `4` instead of `8` here? Perhaps something more dynamic? I can see the `intermediate` struct uses `sizeof()` for missing values, but by passing `8` I think this is overridden. Or have I misunderstood?

Reply:
Thank you, that's much clearer.